keps/sig-instrumentation/1668-trace-context-propagation/README.md

```diff
@@ -0,0 +1,568 @@
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to propagate trace context across components and across a series of related objects originating from an user request. It lays the foundation for enhancing relevant but scattered logs with the trace ID as a common identifier.
+
+## Motivation
+
+Current logging for a series of related messages lacks common identifiers that can be commonly found in other distributed systems. (such as [Global Request IDs](https://specs.openstack.org/openstack/oslo-specs/specs/pike/global-req-id.html)  in openstack) This makes debugging, auditing, reproducing problems, analyzing root cause via logs across components hard, administrators and developers have to match logs by basically using timestamps and object's name as hints which may takes a huge cost especially in a scenario with a large number of requests occur in a short period of time.
+
+### Definitions
+
+**Span**: The smallest unit of a trace.  It has a start and end time, and is attached to a single trace.
+
+**Trace**: A collection of Spans which represents a single process.
+
+**Trace Context**: A reference to a Trace that is designed to be propagated across component boundaries.  Sometimes referred to as the "Span Context".  It is can be thought of as a pointer to a parent span that child spans can be attached to.
+
+### Goals
+
+- Trace context received by the API Server as part of [API Server Tracing](https://github.com/kubernetes/enhancements/issues/647) can be propagated to kubernetes components
```

Comment:
Can you add some example for Kubernetes components (KCM, scheduler, etcd?)

Reply:
[API Server Tracing](https://github.com/kubernetes/enhancements/issues/647) provided the feature for trace context propagation to apiserver, after the trace context has been persisted to etcd, the objects carrying the trace context will do the propagation to other components when they are referenced.

E.g.: kubectl creates a new deployment, there will be a new trace context created in apiserver and persisted to etcd with the object, when controller trying to reconcil the intent, it will get the trace context within the object, which means that be propagated to other components other than apiserver.